mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantsFragment.java

```diff
@@ -0,0 +1,136 @@
+package org.mifos.mobilewallet.mifospay.merchants.ui;
+
+import android.content.res.Resources;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.mifos.mobilewallet.core.data.fineract.entity.accounts.savings.SavingsWithAssociations;
+import org.mifos.mobilewallet.mifospay.R;
+import org.mifos.mobilewallet.mifospay.base.BaseActivity;
+import org.mifos.mobilewallet.mifospay.base.BaseFragment;
+import org.mifos.mobilewallet.mifospay.merchants.MerchantsContract;
+import org.mifos.mobilewallet.mifospay.merchants.adapter.MerchantsAdapter;
+import org.mifos.mobilewallet.mifospay.merchants.presenter.MerchantsPresenter;
+import org.mifos.mobilewallet.mifospay.utils.Constants;
+import org.mifos.mobilewallet.mifospay.utils.RecyclerItemClickListener;
+import org.mifos.mobilewallet.mifospay.utils.Toaster;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class MerchantsFragment extends BaseFragment implements MerchantsContract.MerchantsView {
+
+    @Inject
+    MerchantsPresenter mPresenter;
+    MerchantsContract.MerchantsPresenter mMerchantsPresenter;
+
+    @Inject
+    MerchantsAdapter mMerchantsAdapter;
+
+    @BindView(R.id.inc_state_view)
+    View vStateView;
+    @BindView(R.id.rv_merchants)
+    RecyclerView mRvMerchants;
+    @BindView(R.id.iv_empty_no_transaction_history)
+    ImageView ivTransactionsStateIcon;
+
+    @BindView(R.id.tv_empty_no_transaction_history_title)
+    TextView tvTransactionsStateTitle;
+
+    @BindView(R.id.merchant_fragment_layout)
+    View mMerchantFragmentLayout;
+
+    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
+    TextView tvTransactionsStateSubtitle;
+    private List<SavingsWithAssociations> merchantsList;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_merchants, container, false);
+        ButterKnife.bind(this, rootView);
+        showProgressDialog(Constants.PLEASE_WAIT);
+        mPresenter.attachView(this);
+        mMerchantsPresenter.fetchMerchants();
+        return rootView;
+    }
+
+    private void setupUi() {
+        mMerchantsPresenter.checkMerchants();
+    }
+
+    @Override
+    public void showEmptyStateView() {
+        mMerchantFragmentLayout.setVisibility(View.INVISIBLE);
```

Comment:
Change visibility to "gone" instead of "invisible"

Here's why https://stackoverflow.com/questions/11556607/android-difference-between-invisible-and-gone